protocol/x/clob/types/message_batch_cancel.go

```diff
@@ -0,0 +1,83 @@
+package types
+
+import (
+	errorsmod "cosmossdk.io/errors"
+
+	satypes "github.com/StreamFinance-Protocol/stream-chain/protocol/x/subaccounts/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+var _ sdk.Msg = &MsgBatchCancel{}
+
+// NewMsgBatchCancel constructs a MsgBatchCancel.
+func NewMsgBatchCancel(
+	subaccountId satypes.SubaccountId,
+	cancelBatch []OrderBatch,
+	goodTilBlock uint32,
+) *MsgBatchCancel {
+	return &MsgBatchCancel{
+		SubaccountId:     subaccountId,
+		ShortTermCancels: cancelBatch,
+		GoodTilBlock:     goodTilBlock,
+	}
+}
+
+// ValidateBasic performs stateless validation for the `MsgBatchCancel` msg.
+func (msg *MsgBatchCancel) ValidateBasic() (err error) {
+	subaccountId := msg.GetSubaccountId()
+	if err := subaccountId.Validate(); err != nil {
+		return err
+	}
+
+	cancelBatches := msg.GetShortTermCancels()
+	if len(cancelBatches) == 0 {
+		return errorsmod.Wrapf(
+			ErrInvalidBatchCancel,
+			"Batch cancel cannot have zero orders specified.",
+		)
+	}
+	totalNumberCancels := 0
+	seenClobPairIds := map[uint32]struct{}{}
+	for _, cancelBatch := range cancelBatches {
+		// Check for duplicate clob pair ids across all cancel batches
+		clobPairId := cancelBatch.GetClobPairId()
+		if _, exists := seenClobPairIds[clobPairId]; exists {
```

Comment:
im actually not sure - this is for a subacconut can a subaccount have multiple open perp positions? probably right?

Reply:
actually maybe not im not sure we should look

Reply:
on a second though I actually think you can't because these are strictly for short term orders. Therefore no matter how many short term orders are placed, there will just be a single order for that subaccount that reflects the net change in perp balances of all the short term orders. At least this seems intuitive to me, not sure if this is actually how its handled, we should check